src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralEmail.java

```diff
@@ -256,4 +219,177 @@ private boolean isEmailValid(Applicant applicant) {
             return isNotEmpty(applicant.getEmail());
         }
     }
+
+    void populateSelectedDocsToAttachedList(final CaseData caseData) {
+        if (caseData.getGeneralEmail().getGeScannedDocumentNames() != null
+            && caseData.getGeneralEmail().getGeScannedDocumentNames().getValue().size() > 0) {
+            addSelectedScannedDocuments(caseData);
+        }
+
+        if (caseData.getGeneralEmail().getGeUploadedDocumentNames() != null
+            && caseData.getGeneralEmail().getGeUploadedDocumentNames().getValue().size() > 0) {
+            addSelectedUploadedDocuments(caseData);
+        }
+
+        if (caseData.getGeneralEmail().getGeGeneratedDocumentNames() != null
+            && caseData.getGeneralEmail().getGeGeneratedDocumentNames().getValue().size() > 0) {
+            addSelectedGeneratedDocuments(caseData);
+        }
+
+        if (caseData.getGeneralEmail().getGeApplicant1DocumentNames() != null
+            && caseData.getGeneralEmail().getGeApplicant1DocumentNames().getValue().size() > 0) {
+            addSelectedApp1Documents(caseData);
+        }
+
+        if (caseData.getGeneralEmail().getGeApplicant2DocumentNames() != null
+            && caseData.getGeneralEmail().getGeApplicant2DocumentNames().getValue().size() > 0) {
```

Comment:
If we refactor to 1 or 2 add document methods and pass the collections to the methods as arguments, we could check the presence and size of the collection at the start of the method

Reply:
Checking in common function now.